rl4j/rl4j-core/src/main/java/org/deeplearning4j/rl4j/learning/async/AsyncThreadDiscrete.java

```diff
@@ -81,28 +88,23 @@ protected void preEpoch() {
      * "Subepoch"  correspond to the t_max-step iterations
      * that stack rewards with t_max MiniTrans
      *
-     * @param sObs the obs to start from
-     * @param nstep the number of max nstep (step until t_max or state is terminal)
+     * @param sObs  the obs to start from
+     * @param trainingSteps the number of training steps
      * @return subepoch training informations
      */
-    public SubEpochReturn trainSubEpoch(Observation sObs, int nstep) {
+    public SubEpochReturn trainSubEpoch(Observation sObs, int trainingSteps) {
```

Comment:
This should be NStep size (or the number of steps before we hit episode max steps if smaller), not the epoch/trainingSteps size.

Reply:
This is just different variable naming, the value is min(nstep, remaining episode steps).

Whats a better name here? I could just leave it as nStep, but its not always the value of nstep.

could call it `trainingIterationSteps`?

Reply:
>  the value is min(nstep, remaining episode steps)
Ok that is correct. Sorry, my mistake.

What was wrong with the name `nstep`? That's what is used in litterature....

Reply:
I thought nstep was a bit confusing here because its really a choice between nstep and the number of remaining steps in the episode (whichever is smaller), meaning this argument is not *always* the value of nstep.

I'm happy with nStep though if you feel strongly it should be that?

Reply:
I don't want to impose my will on anything and I would normally agree with that. But I think it's important that people reading our sources be able to easily recognize the algorithms and their different parts from the literature. I think that if a term exists for a concept we should use it.

Reply:
We can talk variable names in the PR that goes through the epoch/episode/trainingStep issues